### PR TITLE
[15.0][MIG] helpdesk_mgmt_project

### PR DESCRIPTION
**Pull request by dalonsod**
Standard migration + fix for search views for Open Tickets filters

### CODE REVIEW

**Comment by dalonsod** [comment]
Hello @jacob88 @manuelregidor could you review? I've added a bugfix in an extra commit, so could be cherry-picked to other Odoo versions.
Thanks!

**Comment by dalonsod** [comment]
Hello @victoralmau @chienandalu could you review? I've added a bugfix in an extra commit, so could be cherry-picked to other Odoo versions.
Thanks!

**Comment by chienandalu** [comment]
Sorry, @dalonsod I don't know this module very well :disappointed:

**Comment by dalonsod** [comment]
> Sorry, @dalonsod I don't know this module very well 😞

Don't worry!

**Comment by dalonsod** [comment]
Hello @ypapouin could you review? I've added a bugfix in an extra commit, so could be cherry-picked to other Odoo versions.
Thanks!

File: helpdesk_mgmt_project/views/project_task_view.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        <field name="model">project.task</field>
+        <field name="inherit_id" ref="project.view_task_form2" />
+        <field name="arch" type="xml">
+            <xpath expr="//div[@name='button_box']" position="inside">
+                <button
+                    class="oe_stat_button"
+                    type="action"
+                    icon="fa-file"
+                    name="%(ticket_action_from_project)d"
+                    context="{'search_default_task_id': id, 'default_task_id': id}"
```
Comment (ypapouin) [inline helpdesk_mgmt_project/views/project_task_view.xml]: The context miss the project:
```
                        context="{
                            'search_default_task_id': id,
                            'default_task_id': id,
                            'search_default_project_id': project_id,
                            'default_project_id': project_id
                        }">
```

File: helpdesk_mgmt_project/models/helpdesk_ticket.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        comodel_name="project.task",
+        compute="_compute_task_id",
+        readonly=False,
+        store=True,
+    )
+
+    @api.depends("project_id")
+    def _compute_task_id(self):
+        for record in self:
+            record.task_id = False
```
Comment (ypapouin) [inline helpdesk_mgmt_project/models/helpdesk_ticket.py]: Before clearing the content, we should check for existing consistent data:
```
    for record in self.filtering('task_id'):
        # Unset a task set by context only if projects mismatch
        if record.task_id.project_id != record.project_id:
            record.task_id = False
```

**Review by ypapouin — CHANGES_REQUESTED** [review]
LGTM, need some other fixing that should be backported

**Comment by dalonsod** [comment]
Thanks @ypapouin comments attended

**Review by ypapouin — APPROVED** [review]
👍

**Review by jacob88 — APPROVED** [review]
LGTM

**Comment by dalonsod** [comment]
@dreispt @pedrobaeza is this PR ready to be merged? Thanks!

**Comment by dalonsod** [comment]
Hello, any news on this? When merged, I'll try to migrate `helpdesk_mgmt_timesheet` as well. Thanks!

**Pull request merged**